monai/apps/deepgrow/interaction.py

```diff
@@ -0,0 +1,81 @@
+# limitations under the License.
+import importlib
+from typing import Dict
+
+import torch
+
+from monai.engines.utils import CommonKeys
+from monai.engines.workflow import Engine, Events
+from monai.transforms import Compose
+
+
+class Interaction:
+    """
+    Deepgrow Training/Evaluation iteration method with interactions (simulation of clicks) support for image and label.
+
+    Args:
+        transforms: execute additional transformation during every iteration (before train).
+            Typically, several Tensor based transforms composed by `Compose`.
+        max_interactions: maximum number of interactions per iteration
+        train: training or evaluation
+        key_probability: field name to fill probability for every interaction
+    """
+
+    def __init__(self, transforms, max_interactions: int, train: bool, key_probability: str = "probability") -> None:
+        self.transforms = transforms
+        self.max_interactions = max_interactions
+        self.train = train
+        self.key_probability = key_probability
+
+        if not isinstance(self.transforms, Compose):
+            transforms = []
+            for t in self.transforms:
+                transforms.append(self.init_external_class(t))
+            self.transforms = Compose(transforms)
+
+    @staticmethod
+    def init_external_class(config_dict):
+        class_args = None if config_dict.get("args") is None else dict(config_dict.get("args"))
+        class_path = config_dict.get("path", config_dict["name"])
+
+        module_name, class_name = class_path.rsplit(".", 1)
+        m = importlib.import_module(module_name)
+        c = getattr(m, class_name)
+        return c(**class_args) if class_args else c()
+
+    def attach(self, engine: Engine) -> None:
+        if not engine.has_event_handler(self, Events.ITERATION_STARTED):
+            engine.add_event_handler(Events.ITERATION_STARTED, self)
+
+    def __call__(self, engine: Engine, batchdata: Dict[str, torch.Tensor]):
+        if batchdata is None:
+            raise ValueError("Must provide batch data for current iteration.")
+
+        for j in range(self.max_interactions):
+            inputs, _ = engine.prepare_batch(batchdata)
+            inputs = inputs.to(engine.state.device)
+
+            engine.network.eval()
+            with torch.no_grad():
+                if engine.amp:
+                    with torch.cuda.amp.autocast():
+                        predictions = engine.inferer(inputs, engine.network)
+                else:
+                    predictions = engine.inferer(inputs, engine.network)
+
+            batchdata.update({CommonKeys.PRED: predictions})
+            batchdata[self.key_probability] = torch.as_tensor(
+                ([1.0 - ((1.0 / self.max_interactions) * j)] if self.train else [1.0]) * len(inputs)
+            )
+            batchdata = self.transforms(batchdata)
+
+        return engine._iteration(engine, batchdata)
```

Comment:
As this handler will attach to `ITERATION_STARTED` and the engine will run `_iteration()` later, why you call `engine._iteration` here?

Reply:
the batchdata is updated... and the _iteration() must happen on new modified batchdata.  will returning batchdata is enough? and _iteration() in later call uses the updated batchdata?  in that case, we need one round of training to make sure it works as expected. 

this code was based on your initial feedback on how to run interactions while training.  may be something got changed recently.  nevertheless, you can confirm what to make _iteration run on new/updated batchdata in above case.

Reply:
Suggest to use:
```py
engine.state.batch = self.transforms(batchdata)
```
Then return nothing in this function.

Thanks.

Reply:
fixed